mozilla-release/mobile/android/base/java/org/mozilla/gecko/tabs/TabsLayoutItemView.java

```diff
@@ -230,5 +247,23 @@ public void setCloseVisible(boolean visible) {
 
     public void setPrivateMode(boolean isPrivate) {
         ((ThemedRelativeLayout) findViewById(R.id.wrapper)).setPrivateMode(isPrivate);
+        /* Cliqz start */
+        ((ThemedLinearLayout) findViewById(R.id.tab_item_header)).setPrivateMode(isPrivate);
+        /* Cliqz end */
+    }
+
+    /* Cliqz start */
+    Drawable getDefaultFavIcon(Tab tab, Context context) {
+            final int drawableId = (tab.isPrivate())? R.drawable.ic_ghosty_forget :
+                    R.drawable.ic_ghosty;
+
+            final Drawable defaultFavIcon = DrawableCompat.wrap(ContextCompat.getDrawable(context,
+                    drawableId)).mutate();
```

Comment:
You are leaking memory here, please cache the two variants and return one of them. This method should ideally be:
```java
Drawable getDefaultFavIcon(Tab tab) {
    if (tab.isPrivate()) {
         return mPrivateDrawable;
    } else {
        return mDrawable;
    }
}
```